crates/wasmtime/src/module.rs

```diff
@@ -593,7 +593,15 @@ impl Module {
     pub fn validate(engine: &Engine, binary: &[u8]) -> Result<()> {
         let mut validator = Validator::new();
         validator.wasm_features(engine.config().features);
-        validator.validate_all(binary)?;
+
+        let mut functions = Vec::new();
```

Comment:
Would it be possible to avoid this intermediate vec and instead directly push each functionbto a worker thread? For example by implementing `rayon::ParallelIterator`?

Reply:
Not easily, no, but it's something we can investigate in the future if necessary.